github/search.go

```diff
@@ -54,6 +55,37 @@ func (s *SearchService) Repositories(query string, opt *SearchOptions) (*Reposit
 	return result, resp, err
 }
 
+// CommitsSearchResult represents the result of a commits search.
+type CommitsSearchResult struct {
+	Total             *int           `json:"total_count,omitempty"`
+	IncompleteResults *bool          `json:"incomplete_results,omitempty"`
+	Commits           []CommitResult `json:"items,omitempty"`
```

Comment:
Again, the current version of not using Items as the name, and not having pointers is consistent with other structs.

However, we can probably deviate on the pointer. What do you think?